base/gmp.jl

```diff
@@ -682,8 +682,16 @@ end
 
 factorial(x::BigInt) = isneg(x) ? BigInt(0) : MPZ.fac_ui(x)
 
+function binomial(n::BigInt, k::Integer)
+    k < 0 && return BigInt(0)
+    k <= typemax(UInt) && return binomial(n, UInt(k))
+    n < 0 && return (isodd(k) ? -one(n) : one(n)) * binomial(k - n - 1, k)
```

Comment:
That just makes an unnecessary allocation and an unnecessary multiplication :-). How about this
```suggestion
    if n < 0
      b = binomial(k - n - 1, k)
      return isodd(k) ? -b : b
    end
```
Or alternatively
```suggestion
    n < 0 && iseven(k) && return binomial(k - n - 1, k)
    n < 0 && isodd(k) && return -binomial(k - n - 1, k)
```